src/Event/Event.jl

```diff
@@ -19,30 +43,43 @@ mutable struct Event{T <: SSDFloat}
     Event{T}() where {T <: SSDFloat} = new{T}(VectorOfArrays(Vector{CartesianPoint{T}}[]), VectorOfArrays(Vector{T}[]), missing, missing)
 end
 
-function Event(location::AbstractCoordinatePoint{T}, energy::RealQuantity = one(T))::Event{T} where {T <: SSDFloat}
-    evt = Event{T}()
-    evt.locations = VectorOfArrays([[CartesianPoint(location)]])
-    evt.energies = VectorOfArrays([[T(to_internal_units(energy))]])
+function Event(location::AbstractCoordinatePoint{T}, energy::RealQuantity = 1.0) where {T <: RealQuantity}
+    pt_internal = to_internal_point(location)
+    T_ = typeof(pt_internal.x)
+    evt = Event{T_}()
+    evt.locations = VectorOfArrays([[pt_internal]])
+    evt.energies  = VectorOfArrays([[T_(to_internal_units(energy))]])
     return evt
 end
```

Comment:
Looks to me like you don't need to define `T` here in the argument list and can then rename `T_` to `T`:
```suggestion
function Event(location::AbstractCoordinatePoint, energy::RealQuantity = 1.0)
    pt_internal = to_internal_point(location)
    T = typeof(pt_internal.x)
    evt = Event{T}()
    evt.locations = VectorOfArrays([[pt_internal]])
    evt.energies  = VectorOfArrays([[T(to_internal_units(energy))]])
    return evt
end
```